docs/docs/command-line-flags.md

```diff
@@ -338,26 +338,65 @@ The Web Site flags control the behavior of Marathon's web site, including the us
 
 ### Metrics Flags
 
-* <span class="label label-default">v0.13.0</span> `--[disable_]metrics` (Optional. Default: enabled):
-    Expose the execution time per method via the metrics endpoint (/metrics) using code instrumentation.
-    Enabling this might noticeably degrade performance but it helps finding performance problems.
-    These measurements can be disabled with --disable_metrics. Other metrics are not affected.
-* <span class="label label-default">v1.6.0</span> `--metrics_averaging_window` (Optional. Default: 30 seconds):
-    Configure the size of the sliding average window that is used to compute the values for the `/metrics` endpoint. Note that this value
-    should be at least double the value of the `kamon.metric.tick-interval` parameter, that is 1 second by default.
-* <span class="label label-default">v0.13.0</span> `--reporter_graphite` (Optional. Default: disabled):
-    Report metrics to [Graphite](http://graphite.wikidot.com) (StatsD) as defined by the given URL.
+* <span class="label label-default">v1.7.0</span> `--metrics_name_prefix`:
+    Configure the prefix that is used when constructing metric names (default: marathon).
+* <span class="label label-default">v1.7.0</span> `--metrics_prometheus`:
+    Enable the StatsD reporter. Once enabled, metrics in the Prometheus
+    format are available at `/metrics/prometheus`.
+* <span class="label label-default">v1.7.0</span> `--metrics_statsd`:
+    Enable the StatsD reporter.
+* <span class="label label-default">v1.7.0</span> `--metrics_statsd_host`:
+    Specify the host to push metrics to in the StatsD format.
+* <span class="label label-default">v1.7.0</span> `--metrics_statsd_port`:
+    Specify the port to push metrics to in the StatsD format.
+* <span class="label label-default">v1.7.0</span> `--metrics_statsd_transmission_interval_ms`:
+    Specify how often to push metrics to a StatsD endpoint (in milliseconds).
+* <span class="label label-default">v1.7.0</span> `--metrics_datadog`:
+    Enable the DataDog reporter.
+* <span class="label label-default">v1.7.0</span> `--metrics_datadog_host`:
+    Specify the host to push metrics to in the DataDog format.
+* <span class="label label-default">v1.7.0</span> `--metrics_datadog_port`:
+    Specify the port to push metrics to in the DataDog format.
+* <span class="label label-default">v1.7.0</span> `--metrics_datadog_protocol`:
```

Comment:
why all the flags?   are we planning to support multiple metrics at the same time?  I don't see the value of that.    I see several better options:
1. `--metrics_host`, `--metrics_port`, `--metrics_transmission_interval_ms` regardless of metric reporter.
2. and/or . `--metric_provider=statsd`

Reply:
Though I do like your suggestion, I am not entirely sure it generalizes well. For instance, what if a user wants to enable multiple reporters? Now, we have Dropwizard (HTTP endpoint), Prometheus (the same), StatsD, DataDog. I bet we'll have more as time goes by. What do you think?

Reply:
there may be different configurations for different impls... but there is a lot of commonality between them as listed.    It would be nice to eliminate all the specific flags and generalize.   the providers we support and how we configure them would then be in documentation.

Reply:
I agree, but I don't see how to get rid of this. Suppose, one wants to enable A and B reporters. There has to be params like `--metrics_a_host` and `--metrics_b_host`, in order to enable them to report to different remote endpoints, as one might expect.